state/executor.go

```diff
@@ -202,6 +209,22 @@ func (t *Transition) Receipts() []*types.Receipt {
 
 var emptyFrom = types.Address{}
 
+func (t *Transition) WriteFailedReceipt(txn *types.Transaction) {
+	receipt := &types.Receipt{
+		CumulativeGasUsed: t.totalGas,
+		TxHash:            txn.Hash,
+		Logs:              t.state.Logs(),
+	}
+
+	receipt.LogsBloom = types.CreateBloom([]*types.Receipt{receipt})
+	receipt.SetStatus(types.ReceiptFailed)
+	t.receipts = append(t.receipts, receipt)
+
+	if msg := txn.Copy(); msg.To == nil {
+		receipt.ContractAddress = crypto.CreateAddress(msg.From, txn.Nonce)
+	}
```

Comment:
`from` in  failed receipt was `0x0000000000000000000000000000000000000000` when I tried locally. Is it the result you expected? And I'm afraid that I didn't understand why you needed to copy transaction here.

Reply:
great catch!
fixed 494f1c81bc95ad91341d66f2550c456445ffd5cd

Reply:
`{"jsonrpc":"2.0","id":1,"result":{"root":"0x0000000000000000000000000000000000000000000000000000000000000000","cumulativeGasUsed":"0x0","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","logs":[],"status":"0x0","transactionHash":"0x45289de9bac0b44062dbd46936b14402095e3995b2800fd06fd72cd6624bb684","transactionIndex":"0x0","blockHash":"0x718f3fd1a2e4f7498e187f69fab013e1398dbfd0cdefabbbd837e12e34b7d27e","blockNumber":"0x4db","gasUsed":"0x0","contractAddress":"0x0000000000000000000000000000000000000000","from":"0x30ea4435167Ee91f9f874b5a894F3282A956C3FF","to":"0x5383Cb489FaCa92365Bb6f9f1FB40bD032E6365d"}}`